cmake/FindNetCDF.cmake

```diff
@@ -1,131 +1,145 @@
-    set(NetCDF_FOUND "${_NetCDF_FOUND}")
-    set(NetCDF_INCLUDE_DIRS "${_NetCDF_INCLUDE_DIRS}")
-    set(NetCDF_LIBRARIES "${_NetCDF_LIBRARIES}")
-    set(NetCDF_VERSION "${_NetCDF_VERSION}")
-
-    include(FindPackageHandleStandardArgs)
-    find_package_handle_standard_args(NetCDF
-      REQUIRED_VARS NetCDF_LIBRARIES
-      # This is not required because system-default include paths are not
-      # reported by `FindPkgConfig`, so this might be empty. Assume that if we
-      # have a library, the include directories are fine (if any) since
-      # PkgConfig reported that the package was found.
-      # NetCDF_INCLUDE_DIRS
-      VERSION_VAR NetCDF_VERSION)
-
-    if (NOT TARGET NetCDF::NetCDF)
-      add_library(NetCDF::NetCDF INTERFACE IMPORTED)
-      set_target_properties(NetCDF::NetCDF PROPERTIES
-        INTERFACE_LINK_LIBRARIES "PkgConfig::_NetCDF")
-    endif ()
-
-    FindNetCDF_get_is_parallel_aware("${_NetCDF_INCLUDEDIR}")
-    # Skip the rest of the logic in this file.
-    return ()
-  endif ()
-endif ()
-
-find_path(NetCDF_INCLUDE_DIR
-  NAMES netcdf.h
-  DOC "netcdf include directories")
-mark_as_advanced(NetCDF_INCLUDE_DIR)
-
-find_library(NetCDF_LIBRARY
-  NAMES netcdf
-  DOC "netcdf library")
-mark_as_advanced(NetCDF_LIBRARY)
-
-if (NetCDF_INCLUDE_DIR)
-  file(STRINGS "${NetCDF_INCLUDE_DIR}/netcdf_meta.h" _netcdf_version_lines
-    REGEX "#define[ \t]+NC_VERSION_(MAJOR|MINOR|PATCH|NOTE)")
-  string(REGEX REPLACE ".*NC_VERSION_MAJOR *\([0-9]*\).*" "\\1" _netcdf_version_major "${_netcdf_version_lines}")
-  string(REGEX REPLACE ".*NC_VERSION_MINOR *\([0-9]*\).*" "\\1" _netcdf_version_minor "${_netcdf_version_lines}")
-  string(REGEX REPLACE ".*NC_VERSION_PATCH *\([0-9]*\).*" "\\1" _netcdf_version_patch "${_netcdf_version_lines}")
-  string(REGEX REPLACE ".*NC_VERSION_NOTE *\"\([^\"]*\)\".*" "\\1" _netcdf_version_note "${_netcdf_version_lines}")
-  set(NetCDF_VERSION "${_netcdf_version_major}.${_netcdf_version_minor}.${_netcdf_version_patch}${_netcdf_version_note}")
-  unset(_netcdf_version_major)
-  unset(_netcdf_version_minor)
-  unset(_netcdf_version_patch)
-  unset(_netcdf_version_note)
-  unset(_netcdf_version_lines)
-
-  FindNetCDF_get_is_parallel_aware("${NetCDF_INCLUDE_DIR}")
-endif ()
-
-include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(NetCDF
-  REQUIRED_VARS NetCDF_LIBRARY NetCDF_INCLUDE_DIR
-  VERSION_VAR NetCDF_VERSION)
-
-if (NetCDF_FOUND)
-  set(NetCDF_INCLUDE_DIRS "${NetCDF_INCLUDE_DIR}")
-  set(NetCDF_LIBRARIES "${NetCDF_LIBRARY}")
-
-  if (NOT TARGET NetCDF::NetCDF)
-    add_library(NetCDF::NetCDF UNKNOWN IMPORTED)
-    set_target_properties(NetCDF::NetCDF PROPERTIES
-      IMPORTED_LOCATION "${NetCDF_LIBRARY}"
-      INTERFACE_INCLUDE_DIRECTORIES "${NetCDF_INCLUDE_DIR}")
-  endif ()
-endif ()
+if(NETCDF_FOUND)
+  # Create NetCDF target
+  add_library(NetCDF SHARED IMPORTED GLOBAL)
+  target_include_directories(NetCDF INTERFACE "${NETCDF_INCLUDE_DIR}" "${NETCDF_CXX_INCLUDE_DIR}")
```

Comment:
```suggestion
  target_include_directories(NetCDF INTERFACE "${NETCDF_INCLUDE_DIR}" "${NETCDF_CXX_INCLUDE_DIR}")
  target_link_libraries(NetCDF INTERFACE "${NETCDF_LIBRARY}" "${NETCDF_CXX_LIBRARY}")
```

Reply:
This seems to address the NetCDF linking issue I was seeing on the UCS system

Reply:
interesting... It seems like it should be redundant with the lines after it, but if it resolved that then 🤷🏾

Reply:
I was seeing it link against one of the libraries, but not both.